### PR TITLE
Support and deprecate empty args

### LINKED ISSUE #572: Deprecation of empty resolver arguments in favor of explicit strings
In OmegaConf 2.0, it is possible to have empty elements in a list of resolver arguments, for instance:
* `${foo:a,}` calls `foo("a", "")`
* `${foo:a,,b}` calls `foo("a", "", "b")`
* `${foo:,}` calls `foo("", "")`

This has been deprecated in OmegaConf 2.1. Users should instead use explicit empty strings, for instance:
* `${foo:a,""}`
* `${foo:a, '', b}`
* `${foo:"", ""}`

### PR DESCRIPTION
**Pull request by odelalleau**
Fixes #572

### CODE REVIEW

File: omegaconf/grammar_visitor.py [hunk 1]
```diff
@@ -246,20 +247,41 @@ def visitSequence(
     ) -> Generator[Any, None, None]:
         from ._utils import _get_value
 
-        assert ctx.getChildCount() >= 1  # element (COMMA element)*
+        # (element (COMMA element?)*) | (COMMA element?)+
+        assert ctx.getChildCount() >= 1
+
+        def empty_str_warning() -> None:
+            txt = ctx.getText()
+            warnings.warn(
```
Comment (omry) [inline omegaconf/grammar_visitor.py:255]: Add a comment:
```
# DEPRECATED: remove in 2.2
```

This is not a practice I follow yet in OmegaConf but I think we should.
it makes it easier to spot those later.

**Review by omry — CHANGES_REQUESTED** [review]
Where are your tests? :)

File: omegaconf/grammar_visitor.py (hunk 1 above)
Comment (odelalleau) [reply omegaconf/grammar_visitor.py:256]: Done in 9ecac27

Also adds the missing tests and news I mentioned in the chat :p

**Review by omry — APPROVED** [review]
lgtm, thanks!

**Pull request merged**